gh_pages/_source/session5/Building-a-Perception-Pipeline.rst

```diff
@@ -132,55 +134,52 @@ Implement Voxel Filter
-
+   Replace ``*cloud_ptr`` with ``*cloud_voxel_filtered`` in the call to ``publishPointCloud(...)`` to publish your new filtered point cloud. Take a look at the function and see how we converted our point cloud to a ROS message in order to publish it.
 
-   .. Note:: If you have the time/patience, I would suggest creating a ros publisher for each type of filter. It is often useful to view the results of multiple filters at once in Rviz and just toggle different clouds.
+   .. Note:: For each type of filter we will create a new publisher. It is often useful to view the results of multiple filters at once in Rviz and just toggle different clouds.
 
-#. Compile
+#. Compile and source
 
    .. code-block:: bash
 
-      catkin build
+      colcon build
+      source install/setup.bash
 
 Viewing Results
 """""""""""""""
-#. Run the (currently small) perception pipeline. Note: In rviz change the global frame to **kinect_link**.
+#. Run the (currently small) perception pipeline (each line should be run in a separate terminal). Note: In rviz change the global frame to **kinect_link**.
 
    .. code-block:: bash
 
-      cd ~
-      roscore
-      rosrun tf2_ros static_transform_publisher 0 0 0 0 0 0 world_frame kinect_link
-      rosrun pcl_ros pcd_to_pointcloud table.pcd 0.1 _frame_id:=kinect_link cloud_pcd:=kinect/depth_registered/points
-      rosrun rviz rviz
-      rosrun lesson_perception perception_node
+      ros2 run tf2_ros static_transform_publisher 0 0 0 0 0 0 world_frame kinect_link
+      ros2 run lesson_perception pcd_to_pointcloud --ros-args -p filename:=~/table.pcd -p tf_frame:=kinect_link -p topic:=/kinect/depth_registered/points
+      ros2 run rviz2 rviz2
+      ros2 launch lesson_perception processing_node.launch.py
 
 #. View results
     
-   Within Rviz, add a *PointCloud2* Display subscribed to the topic "object_cluster". What you see will be the results of the voxel filter overlaid on the original point cloud (assuming you have completed exercise 4.2 and saved a new default config or saved a config for that exercise).
+   Within Rviz, add 2 *PointCloud2* Displays subscribed to the topics "voxel_cluster" and "kinect/depth_registered/points". What you see will be the results of the voxel filter overlaid on the original point cloud.
 
    .. image:: /_static/cloud_voxel_filtered.png
 
 
 #. When you are done viewing the results, try changing the voxel filter size from 0.002 to 0.100 and view the results again.  Reset the filter to 0.002 when done.
 
-   * To see the results of this change, use Ctrl+C to kill the perception node, re-build, and re-run the perception node.
+   * To see the results of this change, use Ctrl+C to kill the perception node, re-build, and re-run the perception node. If you'd like, you can create a new parameter for the voxel filter size. Editing the parameter's default value in the launch file will save you some time as you won't need to re-build before re-launching it!
```

Comment:
I don't think this is true in ROS2?

Or have you instructed them to use --symlink-install?  If so, you might want to make clear that is required.